core/src/services/box/core.rs

```diff
@@ -0,0 +1,377 @@
+                .body(AsyncBody::Empty)
+                .map_err(new_request_build_error)?;
+
+            self.sign(&mut request).await?;
+            let resp = self.client.send(req).await?;
+            let status = resp.status();
+
+            match status {
+                StatusCode::OK => {
+                    let resp_body = &resp.into_body().bytes().await?;
+
+                    let result: BoxListResponse =
+                        serde_json::from_slice(resp_body).map_err(new_json_deserialize_error)?;
+                    let target_type = "folder".to_string();
+                    let target_name = item_name.to_string();
+
+                    if let Some(entry) = result
+                        .entries
+                        .iter()
+                        .find(|entry| entry.item_type == target_type && entry.name == target_name)
+                    {
+                        parent_id = entry.id.clone();
+                    } else {
+                        return Err(Error::new(ErrorKind::Unexpected, &format!("Can't find given folder, Please ensure that the folder path exists and is unique." )));
+                    }
+                }
+                _ => Err(parse_error(resp).await?),
+            }
+        }
+        let mut cache_guard = self.path_cache.lock().await;
+        cache_guard.insert(path, parent_id.clone());
+
+        Ok(parent_id)
+    }
+    pub async fn box_get(&self, path: &str) -> Result<Response<IncomingAsyncBody>> {
+        let url: String = format!(
+            "https://api.box.com/2.0/files/{}/content",
+            self.get_file_id_by_path(path).await?
+        );
+        let mut request = Request::get(&url)
+            .header(CONTENT_TYPE, "application/json")
+            .body(AsyncBody::Empty)
+            .map_err(new_request_build_error)?;
+
+        self.sign(&mut request).await?;
+        self.client.send(request).await
+    }
+
+    pub async fn box_upload(
+        &self,
+        path: &str,
+        size: Option<usize>,
+        content_type: Option<&str>,
+        body: AsyncBody,
+    ) -> Result<Response<IncomingAsyncBody>> {
+        let folder_path=self.truncate_filename(path.trim_end_matches('/'));
+        let file_name="test.txt";
+        let url: String = "https://upload.box.com/api/2.0/files/content".to_string();
+        let mut req = Request::post(&url);
+        req = req.header(CONTENT_TYPE, "multipart/form-data; boundary=my-boundary");
+        let mut req_body = BytesMut::with_capacity(100);
+        write!(
+            &mut req_body,
+            "--my-boundary\ncontent-disposition: form-data; name=\"attributes\"\n\n{{\"name\":\"{}\", \"parent\":{{\"id\":\"{}\"}}}}\n--my-boundary\n",
+            file_name,
+            folder_path// should be folder id
+        ).unwrap();
+
+        write!(
+            &mut req_body,
+            "content-disposition: form-data; name=\"file\"; filename=\"{}\"\n",
```

Comment:
You can use `FormDataPart`'s [`header`](https://opendal.apache.org/docs/rust/opendal/raw/struct.FormDataPart.html#method.header) for this.

For example: 

```rust
let multipart =  Multipart::new()
  .part(FormDataPart::new("attributes")
     .content(attributes_content))
  .part(FormDataPart::new("file")
     .header(CONTENT_DISPOSITION, format!("form-data; name=\"{file}\"; filename=\"{filename}\"").parse().unwrap())
     .content(file_content));

let req: http::request::Builder = Request::post(url);
let req = multipart.apply(req)?;
```

NOTE: FormDataPart doesn't support filename yet, so we have to modify the CONTENT_DISPOSITION by hand for now.

Reply:
> You can use `FormDataPart`'s [`header`](https://opendal.apache.org/docs/rust/opendal/raw/struct.FormDataPart.html#method.header) for this.
> 
> For example:
> 
> ```rust
> let multipart =  Multipart::new()
>   .part(FormDataPart::new("attributes")
>      .content(attributes_content))
>   .part(FormDataPart::new("file")
>      .header(CONTENT_DISPOSITION, format!("form-data; name=\"{file}\"; filename=\"{filename}\"").parse().unwrap())
>      .content(file_content));
> 
> let req: http::request::Builder = Request::post(url);
> let req = multipart.apply(req)?;
> ```
> 
> NOTE: FormDataPart doesn't support filename yet, so we have to modify the CONTENT_DISPOSITION by hand for now. We will fix this later.

thanks, I got you, I check the code before, but haven't think about inserting header again manually, you are right, it's much elegant to implement it.